src/shared/lib/analytics/MixpanelAnalytics.ts

```diff
@@ -1,17 +1,36 @@
 import { Mixpanel } from 'mixpanel-react-native';
 
-import { IAnalyticsService, MixProperties } from './AnalyticsService';
+import { MixProperties } from './AnalyticsService';
 import { APP_VERSION } from '../constants';
 
+export interface IAnalyticsService {
+  track(action: string, payload?: Record<string, any>): void;
+  login(id: string): Promise<void>;
+  logout(): void;
+  init(): Promise<void>;
+  hasInstance(): boolean;
+}
+
 class MixpanelAnalytics implements IAnalyticsService {
-  private mixpanel: Mixpanel;
+  private mixpanel: Mixpanel | null;
+
+  constructor(projectToken: string | undefined) {
```

Comment:
We should not allow creating an instance without a project token as it doesn't make any sense without it

Reply:
ok